src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java

```diff
@@ -0,0 +1,249 @@
+package org.broadinstitute.hellbender.tools.walkers.variantutils;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ExampleProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.VariantWalker;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.SortedSet;
+
+/**
+ * Example/toy program that shows how to implement the VariantWalker interface. Prints supplied variants
+ * along with overlapping reads/reference bases/variants (if present).
+ */
+@CommandLineProgramProperties(
+        summary = "Example tool that prints variants supplied to the specified output file (stdout if none provided), along with overlapping reads/reference bases/variants (if provided)",
+        oneLineSummary = "Example tool that prints variants with optional contextual data",
+        programGroup = ExampleProgramGroup.class,
+        omitFromCommandLine = true
+)
+public final class BlahVetCreation {
+
+    /**
+     * Expected headers for the Variant Table (VET)
+     */
+    public enum HeaderFieldEnum {
+        // TODO is this where the validation step (required vs not) lives  -- fail if there is missing data for a required field
+        // and just leave it empty if not required
+
+        START_POSITION("start_position") { // Required
+             public String getColumnValue(final VariantContext variant) throws IOException {
+                 final String startPosition = String.valueOf(variant.getStart());
+                 if (startPosition.equals(null)) {
+                     throw new IllegalArgumentException("Cannot be missing required value for start_position");
+                 }
+                 return startPosition; // TODO see what James' writer does
+            }
+        },
+
+        REFERENCE_BASES("reference_bases") { // Required
+            public String getColumnValue(final VariantContext variant) throws IOException {
+                final String referenceBase = variant.getReference().getBaseString();
```

Comment:
you dont need to check this either because variant context will fall over.  I should have shown you the vcf spec which would have been helpful to know which required fields you should need to check and which you shouldnt. https://samtools.github.io/hts-specs/VCFv4.2.pdf  if you look at that `1.4.1 Fixed fields` section you'll see which columns are required to have values and which don't.  If they're required by teh spec then you dont need to check them here.